post-processor/vagrant/util.go

```diff
@@ -51,6 +51,23 @@ func CopyContents(dst, src string) error {
 	return nil
 }
 
+// Creates a (hard) link to a file, ensuring that all parent directories also exist.
+func LinkFile(dst, src string) error {
+	dstDir, _ := filepath.Split(dst)
```

Comment:
I think you can just use `filepath.Dir` here

Reply:
The semantics of the two seem to be subtly different in terms of what they return when there is no directly. I fear changing this might break some edge cases - the current logic in there was just lifted straight from the CopyContents function in the same file, so is known to be good.